gtsfm/frontend/correspondence_generator/det_desc_correspondence_generator.py

```diff
@@ -0,0 +1,68 @@
+"""Correspondence generator that utilizes explicit keypoint detection, following by descriptor matching, per image.
+
+Authors: John Lambert
+"""
+
+from typing import Dict, List, Tuple
+
+from dask.delayed import Delayed
+import numpy as np
+
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.feature_extractor import FeatureExtractor
+from gtsfm.frontend.correspondence_generator.correspondence_generator_base import CorrespondenceGeneratorBase
+from gtsfm.frontend.matcher.matcher_base import MatcherBase
+
+
+class QuadraticDetDescCorrespondenceGenerator(CorrespondenceGeneratorBase):
```

Comment:
Can we remove quadratic for the names? Because 1) the number of pairs are determined by the retriever and 2) Its a prefix to both the classes so I dont think it adds value.

If we want to use something to denote pairs, I prefer using `TwoView`.

Also, I think better names could be `CorrespondenceFromSparseFeatures` and `CorrespondenceFromDenseImage`. However, its a personal preference

Reply:
Sure, good catch about the Quadratic-* part being determined by the retriever. I removed that prefix.

I like `CorrespondenceFromSparseFeatures` and `CorrespondenceFromDenseImage`, but they sound a bit more like output products, than the class that computes them.